### PR TITLE
Custom retirement config

### LINKED ISSUE #1368: disable counter retirement for live projects with custom retirement
Linked to https://github.com/zooniverse/Ancient-Lives/issues/18#issuecomment-140395139

Projects like annotate and the ancient lives rebuild will use custom retirement instead of the counter based ones. We should support counters on these live projects but provide a mechanism to disable the retirement from happening via these raw counters.

@ggdhines - re annotate is there any reason to have both custom retirement via aggregation engine and a counter retirement set to a higher number? I.e. retire from aggregation if conditions X otherwise retire from counts if they reach 15?

### PR DESCRIPTION
**Pull request by marten**
Closes #1382
Closes #1368

### CODE REVIEW

File: lib/retirement_schemes/classification_count.rb [hunk 1]
```diff
@@ -1,11 +1,21 @@
...
+      if disabled?
+        false
+      else
+        sw_count.classifications_count >= @count
+      end
     end
 
-    def retire?(count)
-      count.classifications_count >= @count
+    def disabled?
```
Comment (camallen) [inline lib/retirement_schemes/classification_count.rb]: is this used anymore or are we just going to overwrite the criteria scheme to NeverRetire?

Comment (marten) [inline lib/retirement_schemes/classification_count.rb]: No, forgot to remove it after adding NeverRetire.

File: app/models/workflow.rb [hunk 2]
```diff
@@ -67,25 +67,13 @@ def retire_subject(subject_id)
...
-      RetirementSchemes::ClassificationCount.new(*params)
-    else
-      raise StandardError, 'invalid retirement scheme'
-    end
+    criteria = retirement_with_defaults.fetch('criteria')
+    options = retirement_with_defaults.fetch('options')
+    scheme_class = RetirementSchemes.for(criteria).new(options)
   end
 
   def retirement_with_defaults
```
Comment (camallen) [inline app/models/workflow.rb]: I am using this for default rep in the serialzer, https://github.com/zooniverse/Panoptes/blob/f2a92ce3d3e48b3874d29801e3c10e3be9283912/app/serializers/workflow_serializer.rb#L50

**Comment by camallen** [comment]
LGTM

**Pull request merged**